### PR TITLE
Fix helm chart annotations for CRDs installed by Cilium

### PR DESCRIPTION
**Pull request by joestringer**
- install: Use proper crd names in Chart.yaml annotations
- install: Add EgressNATPolicy to Helm charts
- install: Add CiliumEndpointSlices to Helm charts
- install: Lint for missing CRDs in charts

### CODE REVIEW

**Review by christarazi — APPROVED** [review]
Nice, thanks for fixing up.

File: install/kubernetes/cilium/Chart.yaml [hunk 1]
```diff
@@ -24,54 +24,70 @@ annotations:
...
+        in the cluster which are managed by Cilium, including enforcement status,
+        IP addressing and whether the networking is succesfully operational.
+    - kind: CiliumEgressNATPolicy
+      version: v2alpha1
+      name: ciliumegressnatpolicies.cilium.io
+      displayName: Cilium Egress NAT Policy
+      description: |
+        Cilium Endpoint Slice represents the status of groups of pods or nodes
+        in the cluster which are managed by Cilium, including enforcement status,
+        IP addressing and whether the networking is succesfully operational.
```
Comment (kaworu) [inline install/kubernetes/cilium/Chart.yaml]: This description document Cilium Endpoint Slice but it should be about Cilium Egress NAT Policy right?

**Review by kaworu — CHANGES_REQUESTED** [review]
Thanks! LGTM besides the Cilium Egress NAT Policy documentation.

File: install/kubernetes/cilium/Chart.yaml (hunk 1 above)
Comment (joestringer) [reply install/kubernetes/cilium/Chart.yaml]: Huh, I thought I copied this description from somewhere else in the codebase but it looks new to this PR. Thanks for pointing this out, I've written a new description and fixed this up in the relevant commit.

**Review by kaworu — APPROVED** [review]
Thanks @joestringer LGTM now!

**Pull request merged**